### PR TITLE
Restructure download

### PR DESCRIPTION
**Pull request by tempusfrangit**
* Custom Transport (roundtripper)
* Use an errgroup not a waitgroup
* Version Command
* user-agent string

### CODE REVIEW

**Comment by tempusfrangit** [comment]
STACKED COMMIT -> Stacked with  #20 (merge #20 first)

**Comment by anotherjesse** [comment]
I've not yet tested this, but overall this LGTM

File: download/buffer.go [hunk 1]
```diff
@@ -85,88 +73,43 @@ func DownloadFileToBuffer(url string) (*bytes.Buffer, int64, error) {
...
 	elapsed := time.Since(startTime).Seconds()
 	througput := humanize.Bytes(uint64(float64(fileSize) / elapsed))
 	fmt.Printf("Downloaded %s bytes in %.3fs (%s/s)\n", humanize.Bytes(uint64(fileSize)), elapsed, througput)
 
 	buffer := bytes.NewBuffer(data)
 	return buffer, fileSize, nil
 }
+
+func downloadChunk(ctx context.Context, start, end int64, dataSlice []byte, trueURL string) error {
+	client := newClient()
```
Comment (technillogue) [inline download/buffer.go:94]: recreating the client for each chunk always bothered me but we can handle this once we get to downloading multiple files

File: download/buffer.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+		concurrency = maxConcurrency
+		chunkSize = fileSize / int64(concurrency)
+	}
+	if viper.GetBool(optname.Verbose) {
+		fmt.Printf("Downloading %s bytes with %d connections (chunk-size = %s)\n", humanize.Bytes(uint64(fileSize)), concurrency, humanize.Bytes(uint64(chunkSize)))
+	}
+
+	errGroup, ctx := errgroup.WithContext(context.Background())
+
+	data := make([]byte, fileSize)
```
Comment (technillogue) [inline download/buffer.go:65]: we'll need to put a lot of thought into this line

**Review by technillogue — APPROVED** [review]

File: download/buffer.go (hunk 2 above)
Comment (tempusfrangit) [reply download/buffer.go:65]: Part of my plan is to evolve so we have a few new ways to download and we can keep the default as is. This buffer creation bit is just the current way we do things. I def. want something better than "ALLOCATE ALL THE RAM" :)

**Comment by tempusfrangit** [comment]
Tested locally, this appears to work as expected.

**Pull request merged**